bluebrain/repo-patches/packages/neuron/package.py

```diff
@@ -21,434 +20,288 @@ class Neuron(CMakePackage):
-        when="@:8.0.999",
-    )
+    # neuron variants for basic installation
+    variant("interviews", default=False, description="Enable GUI with INTERVIEWS")
+    variant("legacy-unit", default=False, description="Enable legacy units")
+    variant("mpi", default=True, description="Enable MPI parallelism")
+    variant("python", default=True, description="Enable python")
+    variant("tests", default=False, description="Enable building tests")
+    variant("rx3d", default=False, description="Enable cython translated 3-d rxd.", when="+python")
+
+    # variants from coreneuron support
+    variant("coreneuron", default=True, description="Enable CoreNEURON support")
+    variant("gpu", default=False, description="Enable GPU build", when="@9:+coreneuron")
+    variant("openmp", default=False, description="Enable CoreNEURON OpenMP support", when="+coreneuron")
+ 
+    # instrumentation for performance measurement
+    variant("caliper", default=False, description="Add Caliper support")
+
+
+    #################### ALL OF THE BELLOW TO BE DISCUSSED & DELETED #######################
+
+    # add build type 'FastDebug' for result validations
+    # TODO: FastDebug should be removed? Maybe useful for internal tests
+    #       and not BBP CI setup. So should be OK to keep?
     variant(
         "build_type",
         default="RelWithDebInfo",
         description="CMake build type",
         values=("Debug", "FastDebug", "RelWithDebInfo", "Release"),
     )
-    variant("coreneuron", default=True, description="Enable CoreNEURON support")
-    variant(
-        "mod-compatibility",
-        default=True,
-        description="Enable CoreNEURON compatibility for MOD files",
-    )
-    variant("interviews", default=False, description="Enable GUI with INTERVIEWS")
-    variant(
-        "legacy-fr",
-        default=True,
-        description="Use original faraday, R, etc. instead of 2019 nist constants",
-    )
-    variant("memacs", default=True, description="Enable use of memacs")
-    variant("mpi", default=True, description="Enable MPI parallelism")
-    variant("python", default=True, description="Enable python")
-    variant("rx3d", default=True, description="Enable cython translated 3-d rxd.", when="+python")
+
+    # sanitizer to help during development
+    # TODO: Should be removed as it's really only few of core developers
+    #       use it and no need as a general option?
     variant(
         "sanitizers",
         default="None",
-        description="Enable runtime sanitizers",
+        description="Enable runtime LLVM sanitizers",
         multi=True,
         values=("None", "address", "leak", "undefined"),
     )
-    variant("tests", default=False, description="Enable building tests")
+
+    # extended set of model tests
+    # TODO: OK to keep as these can be used by others as generic
+    #       performance tests. But there is still a question in
+    #       my mind if/how anyone else is going to use it and how.
```

Comment:
i would remove this. I think for the special case where you want to run performance tests you'd want to maybe build on the side, doesn't seem like a thing I'd want in a pkg manager.

Reply:
I agree that this shouldn't be in the upstream package. For the BBP CI cases I think this helps a lot to simplify the CIs